src/trace_comparison.rs

```diff
@@ -107,27 +98,41 @@ fn normalize_traces_state_diff(traces: &mut Vec<TransactionTraceWithHash>) {
     }
 }
 
-// Take two JSONs and compare each (key, value) recursively.
-// It will consume the two JSON as well.
-// Store the resutls in an output JSON.
 pub fn generate_block_comparison(
     block_number: u64,
-    base_traces: Vec<TransactionTraceWithHash>,
-    native_traces: Vec<TransactionTraceWithHash>,
+    base_traces: &mut Vec<TransactionTraceWithHash>,
+    native_traces: &mut Vec<TransactionTraceWithHash>,
 ) -> Value {
-    compare_jsons(
-        trace_block_report_to_json(block_number, base_traces),
-        trace_block_report_to_json(block_number, native_traces),
-    )
+    normalize_traces_state_diff(base_traces);
+    normalize_traces_state_diff(native_traces);
+
+    let base_block_report = block_report_with_dependencies(&base_traces);
+    let native_block_report = block_report_with_dependencies(&native_traces);
+
+    let post_response = match (&base_block_report, &native_block_report) {
+        (Value::Array(base), Value::Array(native)) => base
+            .iter()
+            .zip(native.iter())
+            .map(|(base_trace, native_trace)| compare_trace(base_trace, native_trace))
+            .collect(),
+        _ => Vec::new(),
+    };
+
+    json!({
+        "block_num": block_number,
+        "post_response": post_response,
+    })
 }
 
-pub fn compare_jsons(json_1: Value, json_2: Value) -> Value {
+/// Take two JSONs and compare each (key, value) recursively.
+/// Store the results in an output JSON.
+pub fn compare_jsons(json_1: &Value, json_2: &Value) -> Value {
     let output = compare_json_values(json_1, json_2);
     clean_json_value(output)
 }
 
-fn compare_json_values(val_1: Value, val_2: Value) -> Value {
-    match (val_1, val_2) {
```

Comment:
To avoid this clone of Value references would require much more changes to this code. If this is not ideal, then I would propose reverting this commit a06e4be331c646346b56caa42dc2c4cf8b4e7b17 and perform the optimisations in another PR